wire-ios-sync-engine/Tests/Source/UserSession/ZMUserSessionTests+RecurringActions.swift

```diff
@@ -0,0 +1,126 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import XCTest
+@testable import WireSyncEngine
+
+class ZMUserSessionTests_RecurringActions: ZMUserSessionTestsBase {
+
+    var mockRecurringActionService: MockRecurringActionService!
+
+    override func setUp() {
+        super.setUp()
+
+        mockRecurringActionService = MockRecurringActionService()
+    }
+
+    override func tearDown() {
+        mockRecurringActionService = nil
+
+        super.tearDown()
+    }
+
+    func testThatItCallsPerformActionsAfterQuickSync() {
+        // given
+        sut.recurringActionService = mockRecurringActionService
+
+        // when
+        XCTAssertFalse(mockRecurringActionService.performActionsIsCalled)
+        sut.didFinishQuickSync()
+
+        // then
+        XCTAssertTrue(mockRecurringActionService.performActionsIsCalled)
+    }
+
+    func testThatItAddsActions() {
+        // given
+        let action = RecurringAction(id: "11", interval: 5, perform: {})
+        sut.recurringActionService = mockRecurringActionService
+
+        // when
+        XCTAssertEqual(mockRecurringActionService.actions.count, 0)
+        sut.recurringActionService.registerAction(action)
+
+        // then
+        XCTAssertEqual(mockRecurringActionService.actions.count, 1)
+    }
```

Comment:
This is testing the mock recurring action service rather than the user session.

Reply:
Why not test this logic with a mock?

Reply:
I mean to say, the "when" part is not calling a method on sut, but on the recurring action service, which appears to be a mock instance.

Reply:
Yes, you are right. I removed the part with a `sut`